src/libmemcached/man/memcached_strerror.3
.TH "MEMCACHED_STRERROR" "3" "April 03, 2013" "1.0.17" "libmemcached"
.SH NAME
memcached_strerror \- libmemcached Documentation
.
.nr rst2man-indent-level 0
.
.de1 rstReportMargin
\\$1 \\n[an-margin]
level \\n[rst2man-indent-level]
level margin: \\n[rst2man-indent\\n[rst2man-indent-level]]
-
\\n[rst2man-indent0]
\\n[rst2man-indent1]
\\n[rst2man-indent2]
..
.de1 INDENT
.\" .rstReportMargin pre:
. RS \\$1
. nr rst2man-indent\\n[rst2man-indent-level] \\n[an-margin]
. nr rst2man-indent-level +1
.\" .rstReportMargin post:
..
.de UNINDENT
. RE
.\" indent \\n[an-margin]
.\" old: \\n[rst2man-indent\\n[rst2man-indent-level]]
.nr rst2man-indent-level -1
.\" new: \\n[rst2man-indent\\n[rst2man-indent-level]]
.in \\n[rst2man-indent\\n[rst2man-indent-level]]u
..
.\" Man page generated from reStructuredText.
.
.SH SYNOPSIS
.sp
#include <libmemcached/memcached.h>
.INDENT 0.0
.TP
.B const char * memcached_strerror(memcached_st\fI\ *ptr\fP, memcached_return_t\fI\ rc\fP)
.UNINDENT
.sp
Compile and link with \-lmemcached
.SH DESCRIPTION
.sp
\fI\%memcached_strerror()\fP takes a \fBmemcached_return_t\fP value and returns a string describing the error.
.sp
This string must not be modified by the application.
.sp
\fBmemcached_return_t\fP values are returned from nearly all libmemcached(3) functions.
.sp
\fBmemcached_return_t\fP values are of an enum type so that you can set up responses with switch/case and know that you are capturing all possible return values.
.SH RETURN
.sp
\fI\%memcached_strerror()\fP returns a string describing a \fBmemcached_return_t\fP value.
.SH HOME
.sp
To find out more information please check:
\fI\%http://libmemcached.org/\fP
.SH SEE ALSO
.sp
\fImemcached(1)\fP \fIlibmemcached(3)\fP
.SH AUTHOR
Brian Aker
.SH COPYRIGHT
2011-2013, Brian Aker DataDifferential, http://datadifferential.com/
.\" Generated by docutils manpage writer.
.


Changes
=head1 NAME

Memcached::libmemcached::Changes - List of significant changes

=head2 Changes in 1.001702

  Improve handling of case where last_return==MEMCACHED_ERRNO
    but lmc_state->last_errno==0, RT#41299.

=head2 Changes in 1.001701 - 2nd Sept 2013

  Sync with libmemcached 1.0.17
  Fixed links in docs, RT#66291
  Fixed whatis in pod of Memcached::libmemcached::API, RT#71172
  Fixed undef warnings related to expiration, RT#69996
  Don't let DESTDIR env var upset libmemcached build (Chip Salzenberg).

  Note that get_server_for_key() may have a memory leak.

=head2 Changes in 0.4406 4th Jan 2011

  Added get_server_for_key() thanks to Roy Hooper.

=head2 Changes in 0.4405 29th Oct 2010

  Fix crash if memcached_callback_get(MEMCACHED_CALLBACK_PREFIX_KEY) returns NULL.

=head2 Changes in 0.4404 28th Oct 2010

  Added memcached_callback_set() to enable MEMCACHED_CALLBACK_PREFIX_KEY.
  Added memcached_callback_get()
  Made test timing more robust.

=head2 Changes in 0.4403 28th Oct 2010

  Changed memcached_version() to return the lowest server version seen.
  Added new memcached_{inc,dec}rement{_with_initial,}{_by_key,} methods.
  Added new methods memcached_server_add_with_weight() and
    memcached_server_add_unix_socket_with_weight().
  Added new trace_level() method.

=head2 Changes in 0.4402 26th Oct 2010

  Changed walk_stats() API. Now uses the memcached_stat_execute() function
    and no longer passes $stats_arg as the forth argument to the callback.
  Minor updates to docs.

=head2 Changes in 0.4401 21st Oct 2010

  Sync with libmemcached 0.44
  Fixed core dump in stats function.
  Fixed memory leak in stats function. RT#41496
  Improved error reporting in trace log.

=head2 Changes in 0.4202  2nd Aug 2010

  Fix memcached_version() to properly return a list when called in list context
     this should resolve rt #59864

=head2 Changes in 0.4201 16th Jul 2010

  Sync with libmemcached 0.42
  Squashed various compile warnings
  Many updates to libmemcached API calls due to API changes
  Merged existing svn tree (which was out of sync) with 
  Daisuke's tree

=head2 Changes in 0.3101 6th Sep 2009

  Sync with libmemcached 0.31
  Now tested against memcached 1.4.1
  Now tested with binary protocol
  Work with newer libmemcached, which returned MEMCACHED_END instead of
    MEMCACHED_SUCCESS on fetch success

=head2 Changes in 0.2501 (svn r163) 17th Dec 2008

  Sync with libmemcached 0.25.

=head2 Changes in 0.2101 (svn r156) 27th May 2008

  Sync with libmemcached 0.21.
  Report memcached version when running tests.
  Skip flags test when using memcached <1.2.4.
  Test consistent hashing thanks to Daisuke Maki.
  Deprecated undocumented calling get() with an array ref as key
    it now generates a warning.
  Added way to specify multiple memcached servers for tests.
  Added -bin option to Makefile.PL to install mem* utilities
    thanks to Jos Boumans.

=head2 Changes in 0.1902 (svn r144) 17th April 2008

  Fixed leak of $memc which prevented DESTROY (and thus memcached_free)
    being called, thanks to Jim Spath and Daisuke Maki.
  Fixed tests to not use whitespace in keys, thanks to Daisuke Maki.

=head2 Changes in 0.1901 (svn r140) 10th April 2008

  Sync with libmemcached 0.19.
  Added walk_stats method for fetching stats via a callback
    thanks to Daisuke Maki.

=head2 Changes in 0.1701 (svn r129) 3rd March 2008

  Sync with libmemcached 0.17.
  Perl API for memcached_version returns the version info.
  Added auto-generated Memcached::libmemcached::constants docs.
  Improved docs.
  Assorted minor fixes.

=head2 Changes in 0.1405 (svn rXXX)

  Added official object-oriented API for perl-specific extensions.
  Allow SubclassName->new as alias for memcached_create.
  Deprecated perl-specific functions superceeded by methods:
    memcached_errstr                - now $memc->errstr
    memcached_mget_into_hashref     - now $memc->mget_into_hashref
    memcached_set_callback_coderefs - now $memc->set_callback_coderefs
  Added -g/-pg options to Makefile.PL to enable debug/profiling.
  Changed memcache object to be blessed hash instead of blessed int.

=head2 Changes in 0.1404 (svn r106) 29th Jan 2008

  Fixes for perl 5.6.
  Removed dependency on Time::HiRes for testing.

=head2 Changes in 0.1402 (svn r102) 27th Jan 2008

  Added tracing via PERL_LIBMEMCACHED_TRACE env var.
  Added perl callbacks for get and set.
  Added many more functions and improved the docs.
  Reenable perl 5.006 as we supply our own xsubpp now.

=head2 Changes in 0.1401

  Added memcached_errstr and related $memc state management.
  Changes functions to return simple boolean code.
  Added memcached_replace, memcached_cas
  More tests.

=head2 Changes in 0.1308

  Includes libmemcached-0.14
  memcached_return values are now boolean (true, defined false, or undef)
  Added memcached_errstr($memc) to access last error code and string.
  Bundle xsubpp to avoid problems interfacing with our custom ParseXS.

=head2 Changes in 0.1307

  Min perl version now 5.008 due to XS length(other_arg) syntax.

=cut



src/libmemcached/lib/memcached_strerror.c
#include "common.h"

char *memcached_strerror(memcached_st *ptr, memcached_return rc)
{
  switch (rc)
  {
  case MEMCACHED_SUCCESS:
    return "SUCCESS";
  case MEMCACHED_FAILURE:
    return "FAILURE";
  case MEMCACHED_HOST_LOOKUP_FAILURE:
    return "HOSTNAME LOOKUP FAILURE";
  case MEMCACHED_CONNECTION_FAILURE:
    return "CONNECTION FAILURE";
  case MEMCACHED_CONNECTION_BIND_FAILURE:
    return "CONNECTION BIND FAILURE";
  case MEMCACHED_READ_FAILURE:
    return "READ FAILURE";
  case MEMCACHED_UNKNOWN_READ_FAILURE:
    return "UNKNOWN READ FAILURE";
  case MEMCACHED_PROTOCOL_ERROR:
    return "PROTOCOL ERROR";
  case MEMCACHED_CLIENT_ERROR:
    return "CLIENT ERROR";
  case MEMCACHED_SERVER_ERROR:
    return "SERVER ERROR";
  case MEMCACHED_WRITE_FAILURE:
    return "WRITE FAILURE";
  case MEMCACHED_CONNECTION_SOCKET_CREATE_FAILURE:
    return "CONNECTION SOCKET CREATE FAILURE";
  case MEMCACHED_DATA_EXISTS:
    return "CONNECTION DATA EXISTS";
  case MEMCACHED_DATA_DOES_NOT_EXIST:
    return "CONNECTION DATA DOES NOT EXIST";
  case MEMCACHED_NOTSTORED:
    return "NOT STORED";
  case MEMCACHED_STORED:
    return "STORED";
  case MEMCACHED_NOTFOUND:
    return "NOT FOUND";
  case MEMCACHED_MEMORY_ALLOCATION_FAILURE:
    return "MEMORY ALLOCATION FAILURE";
  case MEMCACHED_PARTIAL_READ:
    return "PARTIAL READ";
  case MEMCACHED_SOME_ERRORS:
    return "SOME ERRORS WERE REPORTED";
  case MEMCACHED_NO_SERVERS:
    return "NO SERVERS DEFINED";
  case MEMCACHED_END:
    return "SERVER END";
  case MEMCACHED_DELETED:
    return "SERVER DELETE";
  case MEMCACHED_VALUE:
    return "SERVER VALUE";
  case MEMCACHED_STAT:
    return "STAT VALUE";
  case MEMCACHED_ERRNO:
    return (ptr) ? strerror(ptr->cached_errno) : "SYSTEM ERROR";
  case MEMCACHED_FAIL_UNIX_SOCKET:
    return "COULD NOT OPEN UNIX SOCKET";
  case MEMCACHED_NOT_SUPPORTED:
    return "ACTION NOT SUPPORTED";
  case MEMCACHED_FETCH_NOTFINISHED:
    return "FETCH WAS NOT COMPLETED";
  case MEMCACHED_NO_KEY_PROVIDED:
    return "A KEY LENGTH OF ZERO WAS PROVIDED";
  case MEMCACHED_BUFFERED:
    return "ACTION QUEUED";
  case MEMCACHED_TIMEOUT:
    return "A TIMEOUT OCCURRED";
  case MEMCACHED_MAXIMUM_RETURN:
    return "Gibberish returned!";
  default:
    return "Gibberish returned!";
  };
}
